src/GitVersionCore/Core/Abstractions/IRepositoryMetadataProvider.cs

```diff
@@ -10,37 +10,37 @@ public interface IRepositoryMetadataProvider
         /// <summary>
         /// Find the merge base of the two branches, i.e. the best common ancestor of the two branches' tips.
         /// </summary>
-        Commit FindMergeBase(Branch branch, Branch otherBranch);
-        Commit FindMergeBase(Commit commit, Commit mainlineTip);
-        Commit GetCurrentCommit(Branch currentBranch, string commitId);
-        Commit GetBaseVersionSource(Commit currentBranchTip);
-        List<Commit> GetMainlineCommitLog(Commit baseVersionSource, Commit mainlineTip);
-        IEnumerable<Commit> GetMergeBaseCommits(Commit mergeCommit, Commit mergedHead, Commit findMergeBase);
+        ICommit FindMergeBase(IBranch branch, IBranch otherBranch);
```

Comment:
Since `GitVersionCore` is now moving to become a true core, common to the rest of the GitVersion codebase and only containing abstractions, the namespace `GitVersion.Common`, the folders `Core` and `Abstractions` don't provide much value anymore, imho.

I also feel that the interfaces `IRepositoryMetadataProvider` and `IGitRepository` have overlapping concerns and that it's a bit difficult to separate their intended use, breaking [SRP](https://en.wikipedia.org/wiki/Single-responsibility_principle).

Thoughts?

Reply:
> Since GitVersionCore is now moving to become a true core, common to the rest of the GitVersion codebase and only containing abstractions, the namespace GitVersion.Common, the folders Core and Abstractions don't provide much value anymore, imho.

Yes, I guess we need to review the structure and put all classes under the GitVersion namespace

> I also feel that the interfaces IRepositoryMetadataProvider and IGitRepository have overlapping concerns and that it's a bit difficult to separate their intended use, breaking SRP.

Totaly agree. I added the `IRepositoryMetadataProvider` as a first attempt to refactor the LibGit dependency. Eventually most of the methods should go to the IGitRepository.